src/App.test.js

```diff
@@ -12,3 +19,292 @@ test('renders a reading with the text `Pokédex`', () => {
   const heading = getByText(/Pokédex/i);
   expect(heading).toBeInTheDocument();
 });
+
+test('1 shows the Pokedéx when the route is `/`', () => {
+  const { getByText } = render(
+    <MemoryRouter initialEntries={['/']}>
+      <App />
+    </MemoryRouter>,
+  );
+
+  expect(getByText('Encountered pokémons')).toBeInTheDocument();
+});
+
+function renderWithRouter(
+  ui,
+  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
+) {
+  return {
+    ...render(<Router history={history}>{ui}</Router>),
+    history,
+  };
+}
+
+describe('Render the pokedex with the pokemon', () => {
+  test('2 render one image', () => {
+    const { getAllByText } = renderWithRouter(<App />);
+
+    expect(getAllByText(/Average weight/i).length).toBe(1);
+  });
+});
+
+describe('3 Next pokemon', () => {
+  afterEach(cleanup);
+  const namePokemon = ['Pikachu', 'Charmander', 'Caterpie', 'Ekans',
+    'Alakazam', 'Mew', 'Rapidash', 'Snorlax', 'Dragonair'];
+
+  test('3.1, 3.2 click on the button, appear next pokemon', () => {
+    const { getByText } = renderWithRouter(<App />);
+
+    const nextPokemon = getByText(/Próximo pokémon/i);
+    expect(nextPokemon).toBeInTheDocument();
+
+
+    for (let index = 0; index < namePokemon.length; index += 1) {
```

Comment:
poderia ter feito um .forEach aqui, pra evitar esse for.
```suggestion
namePokemon.forEach((Pokemon) => {
  expect(getByText(Pokemon)).toBeInTheDocument();
  fireEvent.click(nextPokemon);
});
```